src/firefox/index.js

```diff
@@ -47,36 +47,40 @@ export type RemotePortFinderParams = {
 export type RemotePortFinderFn =
   (params?: RemotePortFinderParams) => Promise<number>;
 
-export function defaultRemotePortFinder(
+export async function defaultRemotePortFinder(
   {
     portToTry = REMOTE_PORT,
     retriesLeft = 10,
     connectToFirefox = defaultFirefoxConnector,
   }: RemotePortFinderParams = {}
 ): Promise<number> {
   log.debug(`Checking if remote Firefox port ${portToTry} is available`);
-  function tryToFindAnOpenPort() {
-    return connectToFirefox(portToTry)
-      .then((client) => {
-        log.debug(`Remote Firefox port ${portToTry} is in use ` +
-          `(retries remaining: ${retriesLeft} )`);
-        client.disconnect();
-        portToTry++;
-        if (retriesLeft > 0) {
-          retriesLeft--;
-          return tryToFindAnOpenPort();
-        }
-        else {
-          throw new WebExtError('Too many retries on port search');
-        }
-      })
-      // The connection was refused so this port is good to use.
-     .catch(onlyErrorsWithCode('ECONNREFUSED', () => {
-       return portToTry;
-     }));
+
+  let client;
+
+  while (retriesLeft >= 0) {
+    try {
+      client = await connectToFirefox(portToTry);
+      log.debug(`Remote Firefox port ${portToTry} is in use ` +
+                `(retries remaining: ${retriesLeft} )`);
```

Comment:
while you're at it, you could fix the extra space in the parentheses around `${retriesLeft}` :) no biggie though, just looks odd in the console.